internal/dump/agentpolicies.go

```diff
@@ -0,0 +1,182 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package dump
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"path/filepath"
+
+	"github.com/elastic/elastic-package/internal/common"
+	"github.com/elastic/elastic-package/internal/kibana"
+)
+
+const AgentPoliciesDumpDir = "agent_policies"
+
+type AgentPoliciesDumper struct {
+	name   *string
```

Comment:
When possible, it'd be better to use the zero value to indicate that this is not set, de-referencing pointers is always a risk of panics.
```suggestion
	name   string
```

And then you can check if this is set by checking if `name != ""`.

Reply:
Ok, I'll change to be a regular string.